### PR TITLE
Adding Comunity Model Flavors

### PR DESCRIPTION
**Pull request by mmerce**
Signed-off-by: Mercè Martín Prats <merce@bigml.com>

## Related Issues/PRs

 #5780

## What changes are proposed in this pull request?

Adding documentation on Comunity Model Flavors (as suggested by @dbczumar ) referring to BigMLFlow, new library to implement BigML Models flavor.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

No test needed, as the change does only involve documentation changes. HTML has been generated and looks correct.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Documenting new Comunity Model Flavors: BigMLFlow 

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — APPROVED** [review]
LGTM!

File: docs/source/models.rst [hunk 1]
```diff
@@ -1398,6 +1398,81 @@ For a ``GroupedPmdarima`` model, an example configuration for the ``pyfunc`` ``p
         * If the model is of type ``GroupedProphet``, ``frequency`` as a string type must be provided.
         * If both ``horizon`` and ``n_periods`` are provided with different values.
 
+Comunity Model Flavors
```
Comment (dbczumar) [inline docs/source/models.rst:1401]:
```suggestion
Community Model Flavors
```

File: docs/source/models.rst [hunk 2]
```diff
@@ -1398,6 +1398,81 @@ For a ``GroupedPmdarima`` model, an example configuration for the ``pyfunc`` ``p
...
+----------------------
+
+Other useful MLflow flavors can be found as
+external libraries developed and maintained by Model providers.
+If you are using one of these models, just install their own flavor
+to benefit from the MLFlow capacities. Here's the list of available
+libraries:
```
Comment (dbczumar) [inline docs/source/models.rst:1408]:
```suggestion
Other useful MLflow flavors are developed and maintained by the
MLflow community, enabling you to use MLflow Models with an
even broader ecosystem of machine learning libraries. For more information,
check out the description of each community-developed flavor below.
```

File: docs/source/models.rst [hunk 3]
```diff
@@ -1398,6 +1398,81 @@ For a ``GroupedPmdarima`` model, an example configuration for the ``pyfunc`` ``p
...
+
+
+BigML (``bigmlflow``)
+^^^^^^^^^^^^^^^^^^^^^
+
+The `bigmlflow <https://github.com/bigmlcom/bigmlflow>`_ library implements
+the ``bigml`` model flavor. It enables using
+`BigML supervised models <https://bigml.readthedocs.io/en/latest/local_resources.html>`_
+and offers the ``save_model()``, ``log_model()`` and ``load_model()`` methods
+defined in every MLFlow flavor.
```
Comment (dbczumar) [inline docs/source/models.rst:1422]:
```suggestion
defined in every MLflow flavor.
```

Comment (dbczumar) [inline docs/source/models.rst:1422]:
```suggestion
and offers the ``save_model()``, ``log_model()`` and ``load_model()`` methods.
```

File: docs/source/models.rst [hunk 4]
```diff
@@ -1398,6 +1398,81 @@ For a ``GroupedPmdarima`` model, an example configuration for the ``pyfunc`` ``p
...
+
+Installing bigmlflow
+~~~~~~~~~~~~~~~~~~~~
+
+BigMLFlow can be found in PyPI and is installed by issuing the command:
```
Comment (dbczumar) [inline docs/source/models.rst:1427]:
```suggestion
BigMLFlow can be installed from PyPI as follows:
```

File: docs/source/models.rst [hunk 5]
```diff
@@ -1398,6 +1398,81 @@ For a ``GroupedPmdarima`` model, an example configuration for the ``pyfunc`` ``p
...
+        with open(MODEL_FILE) as handler:
+            model = json.load(handler)
+            bigmlflow.log_model(model,
+                                artifact_path="model",
+                                registered_model_name="my_model")
+
+These methods also add the ``python_function`` flavor to the MLflow Models
+that they produce, allowing the models to be interpreted as generic Python
+functions for inference via :py:func:`mlflow.pyfunc.load_model()`.
+This loaded PyFunc model can only be scored with DataFrame input.
```
Comment (dbczumar) [inline docs/source/models.rst:1458]:
```suggestion
This loaded PyFunc model can only be scored with DataFrame inputs.
```

**Review by dbczumar — APPROVED** [review]
LGTM with a few tiny suggestions - very excited to merge this once they're addressed! Thanks so much, @mmerce !

**Pull request merged**